test/e2e_kubeadm/kubeadm_config_test.go

```diff
@@ -0,0 +1,117 @@
+	authv1 "k8s.io/api/authorization/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	kubeadmConfigName                             = "kubeadm-config"
+	kubeadmConfigRoleName                         = "kubeadm:nodes-kubeadm-config"
+	kubeadmConfigRoleBindingName                  = kubeadmConfigRoleName
+	kubeadmConfigClusterConfigurationConfigMapKey = "ClusterConfiguration"
+	kubeadmConfigClusterStatusConfigMapKey        = "ClusterStatus"
+)
+
+var (
+	kubeadmConfigConfigMapResource = &authv1.ResourceAttributes{
+		Namespace: kubeSystemNamespace,
+		Name:      kubeadmConfigName,
+		Resource:  "configmaps",
+		Verb:      "get",
+	}
+)
+
+// Define container for all the test specification aimed at verifying
+// that kubeadm creates the cluster-info ConfigMap, that it is properly configured
+// and that all the related RBAC rules are in place
+var _ = KubeadmDescribe("kubeadm-config ConfigMap", func() {
+
+	// Get an instance of the k8s test framework
+	f := framework.NewDefaultFramework("kubeadm-config")
+
+	// Tests in this container are not expected to create new objects in the cluster
+	// so we are disabling the creation of a namespace in order to get a faster execution
+	f.SkipNamespaceCreation = true
+
+	It("should exist and be properly configured", func() {
+		cm := GetConfigMap(f.ClientSet, kubeSystemNamespace, kubeadmConfigName)
+
+		Expect(cm.Data).To(HaveKey(kubeadmConfigClusterConfigurationConfigMapKey))
+		Expect(cm.Data).To(HaveKey(kubeadmConfigClusterStatusConfigMapKey))
+
+		m := unmarshalYaml(cm.Data[kubeadmConfigClusterStatusConfigMapKey])
+		if _, ok := m["apiEndpoints"]; ok {
+			d := m["apiEndpoints"].(map[interface{}]interface{})
+			// get all master nodes
+			masters := GetMasterNodes(f.ClientSet)
+
+			// checks that all the master nodes are in the apiEndpoints list
+			for _, master := range masters.Items {
+				if _, ok := d[master.Name]; !ok {
+					framework.Failf("failed to get apiEndpoints for master %s in %s", master.Name, kubeadmConfigClusterStatusConfigMapKey)
+				}
+			}
+		} else {
+			framework.Failf("failed to get apiEndpoints from %s", kubeadmConfigClusterStatusConfigMapKey)
+		}
+	})
+
+	It("should have related Role and RoleBinding", func() {
+		ExpectRole(f.ClientSet, kubeSystemNamespace, kubeadmConfigRoleName)
+		ExpectRoleBinding(f.ClientSet, kubeSystemNamespace, kubeadmConfigRoleBindingName)
+	})
+
+	It("should be accessible for bootstrap tokens", func() {
+		ExpectSubjectHasAccessToResource(f.ClientSet,
+			rbacv1.GroupKind, bootstrapTokensGroup,
+			kubeadmConfigConfigMapResource,
+		)
+	})
+
+	It("should be accessible for for nodes", func() {
+		ExpectSubjectHasAccessToResource(f.ClientSet,
+			rbacv1.GroupKind, nodesGroup,
+			kubeadmConfigConfigMapResource,
+		)
+	})
+})
+
+func GetClusterConfiguration(c clientset.Interface) map[interface{}]interface{} {
```

Comment:
if this function is `public` shouldn't go to the `util` file? :thinking:

Reply:
switched to private